fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMultiTargetTest.java

```diff
@@ -775,4 +777,153 @@ public void benchMarkMultiTarget() {
         System.out.printf("%d indexes, %d records. Single build took %d milliSeconds, MultiIndex took %d%n",
                 indexes.size(), numRecords, endSingle - startSingle, endMulti - startMulti);
     }
+
+    @Test
+    void testMultiTargetIndexingBlocker() {
+        // block indexing, then release
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final int numRecords = 107;
+        final int chunkSize  = 17;
+
+        List<Index> indexes = new ArrayList<>();
+        indexes.add(new Index("indexD", new GroupingKeyExpression(EmptyKeyExpression.EMPTY, 0), IndexTypes.COUNT));
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexB", field("num_value_3_indexed"), IndexTypes.VALUE));
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+
+        openSimpleMetaData();
+        populateData(numRecords);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        // 1. partly build multi
+        buildIndexAndCrashHalfway(chunkSize, 2, timer, newIndexerBuilder()
```

Comment:
Is there any way to test aborting a build?
I think in order to do that, record-layer would need infrastructure to have longer-running tests, but if you know of a way that would be great. Perhaps it could be done with the config loader?

Reply:
Added `testMutualIndexingBlockerOnTheFly`.